test_conformance/api/test_clone_kernel.cpp

```diff
@@ -324,78 +453,471 @@ REGISTER_TEST_VERSION(clone_kernel, Version(2, 1))
+        "argument.");
+
+    test_assert_error(
+        ((int*)pbufRes)[6] == write_val,
+        "clCloneKernel test failed.  Failed to clone cl_mem argument.");
+
+    if (bimg)
     {
-        test_error( error, "clCloneKernel test failed. Failed to clone integer type argument." );
-        return -1;
+        error = test_image_arg_shallow_clone(device, context, queue,
+                                             num_elements, pbufRes, bufOut);
+        test_error(error, "image arg shallow clone test failed.");
     }
 
-    if (((int*)pbufRes)[1] != sa.i)
+    if (bdouble)
     {
-        test_error( error, "clCloneKernel test failed. Failed to clone structure type argument." );
-        return -1;
+        error = test_double_arg_clone(device, context, queue, num_elements,
+                                      pbufRes, bufOut);
+        test_error(error, "double arg clone test failed.");
+    }
+
+    return TEST_PASS;
+}
+
+int test_svm_enqueue_helper(cl_context context, cl_command_queue queue,
+                            cl_int* svmPtr_Kernel, cl_kernel srcKernel,
+                            cl_int value)
+{
+    cl_int error;
+    size_t ndrange1 = 1;
+
+    // enqueue - srcKernel
+    error = clEnqueueNDRangeKernel(queue, srcKernel, 1, NULL, &ndrange1, NULL,
+                                   0, NULL, NULL);
+    test_error(error, "clEnqueueNDRangeKernel failed");
+    error = clFinish(queue);
+
+    test_error(error, "clFinish failed");
+    error = clEnqueueSVMMap(queue, CL_TRUE, CL_MAP_READ | CL_MAP_WRITE,
+                            svmPtr_Kernel, sizeof(cl_int), 0, NULL, NULL);
+    test_error(error, "clEnqueueSVMMap failed");
+    test_assert_error(svmPtr_Kernel[0] == value,
+                      "clCloneKernel test failed, Failed to verify "
+                      "integer value from SVM pointer. ");
+
+    error = clEnqueueSVMUnmap(queue, svmPtr_Kernel, 0, NULL, NULL);
+    test_error(error, "clEnqueueSVMUnmap failed");
+    error = clFinish(queue);
+    test_error(error, "clFinish failed");
+
+    return TEST_PASS;
+}
+
+int test_svm_exec_info_helper(cl_context context, cl_command_queue queue,
+                              BufPtr* pBuf, cl_int* svmPtr_Kernel,
+                              cl_kernel srcKernel, cl_int value)
+{
+    cl_int error;
+
+    error = clEnqueueSVMMap(queue, CL_TRUE, CL_MAP_READ | CL_MAP_WRITE, pBuf,
+                            sizeof(BufPtr), 0, NULL, NULL);
+    test_error(error, "clEnqueueSVMMap failed");
+
+    pBuf->store = svmPtr_Kernel;
+
+    error = clEnqueueSVMUnmap(queue, pBuf, 0, NULL, NULL);
+    test_error(error, "clEnqueueSVMUnmap failed");
+    error = clFinish(queue);
+    test_error(error, "clFinish failed");
+
+    error = clSetKernelArg(srcKernel, 0, sizeof(cl_int), &value);
+    test_error(error, "clSetKernelArg failed");
+    error = clSetKernelArgSVMPointer(srcKernel, 1, pBuf);
+    test_error(error, "clSetKernelArgSVMPointer failed");
```

Comment:
This is called `test_svm_exec_info_helper` but I don't see any place where `clSetKernelExecInfo` is called.

Since we're passing `svmPtr_Kernel` indirectly, I think we need something like:


```suggestion
    error = clSetKernelArgSVMPointer(srcKernel, 1, pBuf);
    test_error(error, "clSetKernelArgSVMPointer failed");

    error = clSetKernelExecInfo(srcKernel, CL_KERNEL_EXEC_INFO_SVM_PTRS,
                                sizeof(svmPtr_Kernel), &svmPtr_Kernel);
    test_error(error, "clSetKernelExecInfo failed");
```

Reply:
Done.